### ISSUE TITLE
EDA

### ISSUE THREAD

**Issue by sidhu1012** [Dacience/DataScientist#29]
Create a module for EDA which takes features and targets as input

**Comment by parva-jain** [comment]
I want to create a class for EDA having many methods performing different analyses. How would be it?

**Comment by sidhu1012** [comment]
> I want to create a class for EDA having many methods performing different analyses. How would be it?

It would be good. Can you please explain what you exactly have in your mind.

**Comment by parva-jain** [comment]
So basically a class named EDA having many functions and each function will be performing some sort of analysis. 
For example, one function will preview data along with its shape, then one function will return the columns with missing values along with the count, then one function will plot the distribution plots of the mentioned columns, similarly pair plots and followed by other EDA techniques. So that's my rough plan.

**Comment by sidhu1012** [comment]
> So basically a class named EDA having many functions and each function will be performing some sort of analysis.
> For example, one function will preview data along with its shape, then one function will return the columns with missing values along with the count, then one function will plot the distribution plots of the mentioned columns, similarly pair plots and followed by other EDA techniques. So that's my rough plan.

Actually my plan is to define a function which does all of this in one call, because calling it step by step i.e by a different function call from class won't be beneficial.

**Comment by parva-jain** [comment]
but there is so much in EDA and there are so many choices. Is that efficient if we output pair plot, scatter plot, distplot, box plot, histograms, violin plot, and many more just under one call?

**Comment by sidhu1012** [comment]
> but there is so much in EDA and there are so many choices. Is that efficient if we output pair plot, scatter plot, distplot, box plot, histograms, violin plot, and many more just under one call?

Yes ,if all is not under one head then why would it be beneficial and why someone would use it rather than calling every plot themselves.

**Comment by parva-jain** [comment]
I have no issue implementing like the way u r saying but still, I believe that if we make EDA like this, then it will reduce the human insight on the data, and basically, the one using it will not bother about the knowledge of why he wants to look upon a box plot or histogram as she will get all plots just by calling EDA and that's where it's contrary to the name 'Exploratory Data Analysis' as exploration will diminish. So just the last explanation on this topic and will go with ur final call.

**Comment by parva-jain** [comment]
@sidhu1012 what's the final take on this?

**Comment by sidhu1012** [comment]
> @sidhu1012 what's the final take on this?

I think all under one head is final. Things can change according to need later.

**Comment by parva-jain** [comment]
Alright....will do it.

**Comment by parva-jain** [comment]
hey @sidhu1012, there are some different EDA techniques for numerical and categorical features so can I take input as numerical features, categorical features, and target since dynamically separating features into numerical and categorical won't be correct on certain types of datasets.

**Comment by sidhu1012** [comment]
> hey @sidhu1012, there are some different EDA techniques for numerical and categorical features so can I take input as numerical features, categorical features, and target since dynamically separating features into numerical and categorical won't be correct on certain types of datasets.

Design 3 different functions for every type of features and a master function which calls the appropriate one according to user's choice.
